pyomo/gdp/tests/test_hull.py

```diff
@@ -108,16 +108,16 @@ def check_furman_et_al_denominator(self, expr, ind_var):
 
     def test_transformed_constraint_nonlinear(self):
         m = models.makeTwoTermDisj_Nonlinear()
-        TransformationFactory('gdp.hull').apply_to(m)
+        hull = TransformationFactory('gdp.hull')
+        hull.apply_to(m)
 
         disjBlock = m._pyomo_gdp_hull_reformulation.relaxedDisjuncts
 
         # the only constraint on the first block is the non-linear one
-        disj1c = disjBlock[0].component("d[0].c")
-        self.assertIsInstance(disj1c, Constraint)
+        disj1c = hull.get_transformed_constraints(m.d[0].c)
         # we only have an upper bound
         self.assertEqual(len(disj1c), 1)
-        cons = disj1c['ub']
+        cons = disj1c[0]
```

Comment:
So the order of the constraints is `0`: `ub`, `1`: `lb`? Would this break the older code if we were looking for the transformed upper and lower bounds constraints using the `ub`/`lb` keys?

Reply:
I think that the performance improvement justifies the breaking of older code. As long as documentation reports this and if from the error it is easy to know when the actual syntax is supposed to be I fully support these changes.